CMakeLists.txt

```diff
@@ -2,6 +2,11 @@ cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
 
 project(RigelEngine)
 
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m32")
```

Comment:
I'm unsure if this flag is actually necessary. When all the external dependencies are coming from Emscripten, and the other ones are built as part of the build, I would expect things to work without this. Could you give it a try?

Reply:
Yeah, it is not. Though I would like to keep it in some form because wasm64 doesn't exist AFAIK, for now. So for emscripten, I would like to make sure it builds only 32-bit

Reply:
That should be the responsibility of the Emscripten toolchain file, however. The general philosophy of CMake is that Toolchain files are used to define flags specific to the target system. Project files (`CMakeListst.txt`) should ideally not contain any compiler flags at all. It's hard to achieve in practice (unless you would write and maintain your own toolchain files), but I would still like to follow it where possible.

The idea is that if wasm64 would be introduced one day, we would only have to update Emscripten, but not change any code in our `CMakeLists`. 

Take a look at [this line](https://github.com/emscripten-core/emscripten/blob/efe15600fc2d9210f4a151080f3eab76eec7499c/cmake/Modules/Platform/Emscripten.cmake#L26): The Emscripten toolchain file tells CMake that the platform is a 32-bit one.